core/src/main/kotlin/com/justai/jaicf/generic/TypeToken.kt

```diff
@@ -0,0 +1,64 @@
+package com.justai.jaicf.generic
+
+import com.justai.jaicf.api.BotRequest
+import com.justai.jaicf.context.ActivatorContext
+import com.justai.jaicf.reactions.Reactions
+import kotlin.reflect.KClass
+
+/**
+ * Type token that holds information about certain types of [ActivatorContext], [BotRequest] and [Reactions].
+ * Can be used in some contexts in order to provide type-specific functionality.
+ * Can be composed with other type tokens, see [and] functions.
+ */
+data class ContextTypeToken<A: ActivatorContext, B: BotRequest, R: Reactions>(
+    val activatorType: KClass<A>,
```

Comment:
Could we use `internal` modifier here to hide these params in user's scenarios?

Reply:
It might be useful to hide these parameters from scenarios, but I definitely don't want to make it internal, as it will limit us and a user in developing some new features based on these tokens.

Reply:
What kind of features do you mean?